### PR TITLE
Added new module for supporting DNS recordset operations in Private DNS zone and Resolved issue #265

### LINKED ISSUE #265: azure_rm_dnsrecordset does not support private dnszones
##### SUMMARY
`azure_rm_dnsrecordset` Does not support the creating of records in private dns zones. When you try to add records with `azure_rm_dnsrecordset` you get a error: `The Resource 'Microsoft.Network/dnszones/<zone_name>' under resource group '<resource_group>' was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix\n"`. Since there is no other supported option to add records to private dns zones and this was supported before this is unwanted functionality.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

`azure_rm_dnsrecordset`

##### ANSIBLE VERSION

```paste below
2.9.13
```

##### CONFIGURATION

```paste below

```

##### OS / ENVIRONMENT

OSX Catalina, python 3.7, pip 20.2.2. 

##### STEPS TO REPRODUCE

- Try to add records with `azure_rm_dnsrecordset` to a private dns zone.

```yaml

```

##### EXPECTED RESULTS

Succesfull recordset creation within a private dnszone

##### ACTUAL RESULTS

```paste below
`The Resource 'Microsoft.Network/dnszones/<zone_name>' under resource group '<resource_group>' was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix\n"`
```

##### POSSIBLE SOLUTION
I think the API is called at Microsoft.Network/dnszones and should be called at Microsoft.Network/privateDnsZones for private dns zones. See Azure API doc here: https://docs.microsoft.com/en-us/rest/api/dns/privatedns/recordsets/createorupdate

### PR DESCRIPTION
**Pull request by aparna-patil**
**SUMMARY**
This module introduces following DNS record sets operations for a Private DNS zone.
1. Create DNS record set in a Private DNS zone
2. Update DNS record set in a Private DNS zone
3. Delete DNS record set in a Private DNS zone
4. Get DNS record set details from Private DNS zone

Fixes #265 

**ISSUE TYPE**
- New Module Pull Request

**COMPONENT NAME**
azure_rm_privatednsrecordset

**ADDITIONAL INFORMATION**
This module allows to perform the create/update, delete and list operations on DNS record sets in an existing Private DNS zone. Azure API doc - https://docs.microsoft.com/en-us/rest/api/dns/privatedns/recordsets/createorupdate

### CODE REVIEW

**Comment by aparna-patil** [comment]
Hi @Fred-sun, Could you please review this PR and provide your feedback? Thank you!

**Comment by Fred-sun** [comment]
> Hi @Fred-sun, Could you please review this PR and provide your feedback? Thank you!

I will start reviewing this soon! Thank you!

File: tests/integration/targets/azure_rm_privatednsrecordset/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+      - entry:
+          - "bar"
+          - "baz"
+  register: results
+
+- name: Assert that TXT record set was created
+  assert:
+    that:
+      - results.changed
+
```
Comment (Fred-sun) [inline tests/integration/targets/azure_rm_privatednsrecordset/tasks/main.yml:214]: Please add test cases and get information through azure_rm_privatednsrecordset_info, thank you

**Review by Fred-sun — COMMENTED** [review]
None of the parameters in the document have defined types!
"return" in the return value should be changed to "returned"！
The 72, 92, 125 lines of azure_rm_privatednsrecordset_info are repeated!

File: plugins/modules/azure_rm_privatednsrecordset_info.py [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+DOCUMENTATION = '''
+---
+module: azure_rm_privatednsrecordset_info
+
+version_added: "1.1.0"
+
+short_description: Get Private DNS Record Set facts
+
+description:
+    - Get facts for a specific DNS Record Set in a Private DNS Zone, or a specific type of DNS record in all zones or
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_privatednsrecordset_info.py]:
```suggestion
    - Get facts for a specific DNS Record Set in a Private DNS Zone, or a specific type of DNS record in all zones or one zone etc.
```

**Review by Fred-sun — COMMENTED** [review]
Please add the following two lines in ‘tests/sanity/ignore-2.10.txt’ and ’tests/sanity/ignore-2.11.txt‘!
```e.g.
plugins/modules/azure_rm_privatednsrecordset.py validate-modules:doc-elements-mismatch
plugins/modules/azure_rm_privatednsrecordset.py validate-modules:invalid-ansiblemodule-schema

File: plugins/modules/azure_rm_privatednsrecordset_info.py (hunk 2 above)
Comment (aparna-patil) [reply plugins/modules/azure_rm_privatednsrecordset_info.py]: Hi @Fred-sun The text on line no. 23 is continued on line no. 24th due to PEP8 standards. 
24 :       one zone etc.

File: tests/integration/targets/azure_rm_privatednsrecordset/tasks/main.yml (hunk 1 above)
Comment (aparna-patil) [reply tests/integration/targets/azure_rm_privatednsrecordset/tasks/main.yml:214]: Added the test cases to get the information for dnsrecordsets through azure_rm_privatednsrecordset_info, Thank you!

**Comment by aparna-patil** [comment]
> Please add the following two lines in ‘tests/sanity/ignore-2.10.txt’ and ’tests/sanity/ignore-2.11.txt‘!
> 
> ```
> plugins/modules/azure_rm_privatednsrecordset.py validate-modules:doc-elements-mismatch
> plugins/modules/azure_rm_privatednsrecordset.py validate-modules:invalid-ansiblemodule-schema
> ```

Done, Added these 2 lines in ‘tests/sanity/ignore-2.10.txt’ and ’tests/sanity/ignore-2.11.txt‘. Thank you

**Comment by aparna-patil** [comment]
> None of the parameters in the document have defined types!
> "return" in the return value should be changed to "returned"！
> The 72, 92, 125 lines of azure_rm_privatednsrecordset_info are repeated!

@Fred-sun , Done, Added the changes as suggested, Please review again and let me know if any other changes are required, Thank you!

File: plugins/modules/azure_rm_privatednsrecordset_info.py [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_privatednsrecordset_info.py]:
```suggestion
    - Get facts for a specific DNS Record Set in a Private DNS Zone, or a specific type of DNS record in all zones or
```

**Comment by aparna-patil** [comment]
Hi @Fred-sun, Kindly let me know if any other changes required in this PR? I have incorporated all the previous suggested changes. could you please take a look at it once and share your feedback? Thank you!

File: plugins/modules/azure_rm_privatednsrecordset_info.py (hunk 3 above)
Comment (aparna-patil) [reply plugins/modules/azure_rm_privatednsrecordset_info.py]: Hi @Fred-sun The text on line no. 23 is continued on line no. 24th due to PEP8 standards.
24 : one zone etc.

Comment (aparna-patil) [reply plugins/modules/azure_rm_privatednsrecordset_info.py]: @Fred-sun Sorry, my bad! I have committed the changes. Thank you!

**Comment by Fred-sun** [comment]
@aparna-patil There are still many formatting errors in the document! Please refer to the link ---"https://docs.ansible.com/ansible/latest/dev_guide/testing_integration.html", the command is --"ansible-test sanity --color -v --junit”
```e.g.
ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:23:118: W291: trailing whitespace
ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: deprecation-mismatch: "meta/runtime.yml" and DOCUMENTATION.deprecation do not agree.
ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: doc-choices-do-not-match-spec: Argument 'auth_source' in argument_spec defines choices as (['auto', 'cli', 'env', 'credential_file', 'msi']) but documentation defines choices as ([])
ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: doc-choices-do-not-match-spec: Argument 'cert_validation_mode' in argument_spec defines choices as (['validate', 'ignore']) but documentation defines choices as ([])
ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: doc-default-does-not-match-spec: Argument 'api_profile' in argument_spec defines default as ('latest') but documentation defines default as (None)
ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: doc-default-does-not-match-spec: Argument 'append_tags' in argument_spec defines default as (True) but documentation defines default as (False)
ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: doc-default-does-not-match-spec: Argument 'auth_source' in argument_spec defines default as ('auto') but documentation defines default as (None)
ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: doc-default-does-not-match-spec: Argument 'cloud_environment' in argument_spec defines default as ('AzureCloud') but documentation defines default as (None)
ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'ad_user' in argument_spec defines type as 'str' but documentation doesn't define type
ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'adfs_authority_url' in argument_spec defines type as 'str' but documentation doesn't define type
ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'api_profile' in argument_spec defines type as 'str' but documentation doesn't define type
ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'append_tags' in argument_spec defines type as 'bool' but documentation doesn't define type
ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'auth_source' in argument_spec defines type as 'str' but documentation doesn't define type
ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'cert_validation_mode' in argument_spec defines type as 'str' but documentation doesn't define type
ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'client_id' in argument_spec defines type as 'str' but documentation doesn't define type
ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'cloud_environment' in argument_spec defines type as 'str' but documentation doesn't define type
ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'password' in argument_spec defines type as 'str' but documentation doesn't define type
ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'profile' in argument_spec defines type as 'str' but documentation doesn't define type
ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'record_type' in argument_spec defines type as 'str' but documentation doesn't define type
ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'relative_name' in argument_spec defines type as 'str' but documentation doesn't define type
ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'resource_group' in argument_spec defines type as 'str' but documentation doesn't define type
ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'secret' in argument_spec defines type as 'str' but documentation doesn't define type
ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'subscription_id' in argument_spec defines type as 'str' but documentation doesn't define type
ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'tags' in argument_spec defines type as 'dict' but documentation doesn't define type
ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'tenant' in argument_spec defines type as 'str' but documentation doesn't define type
ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'top' in argument_spec defines type as 'int' but documentation doesn't define type
ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'zone_name' in argument_spec defines type as 'str' but documentation doesn't define type
ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:43:5: documentation-syntax-error: DOCUMENTATION is not valid YAML
ERROR: tests/sanity/ignore-2.10.txt:725:1: A100: Ignoring 'doc-elements-mismatch' on 'plugins/modules/azure_rm_privatednsrecordset.py' is unnecessar

**Comment by aparna-patil** [comment]
> @aparna-patil There are still many formatting errors in the document! Please refer to the link ---"https://docs.ansible.com/ansible/latest/dev_guide/testing_integration.html", the command is --"ansible-test sanity --color -v --junit”
> 
> ```
> ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:23:118: W291: trailing whitespace
> ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: deprecation-mismatch: "meta/runtime.yml" and DOCUMENTATION.deprecation do not agree.
> ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: doc-choices-do-not-match-spec: Argument 'auth_source' in argument_spec defines choices as (['auto', 'cli', 'env', 'credential_file', 'msi']) but documentation defines choices as ([])
> ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: doc-choices-do-not-match-spec: Argument 'cert_validation_mode' in argument_spec defines choices as (['validate', 'ignore']) but documentation defines choices as ([])
> ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: doc-default-does-not-match-spec: Argument 'api_profile' in argument_spec defines default as ('latest') but documentation defines default as (None)
> ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: doc-default-does-not-match-spec: Argument 'append_tags' in argument_spec defines default as (True) but documentation defines default as (False)
> ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: doc-default-does-not-match-spec: Argument 'auth_source' in argument_spec defines default as ('auto') but documentation defines default as (None)
> ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: doc-default-does-not-match-spec: Argument 'cloud_environment' in argument_spec defines default as ('AzureCloud') but documentation defines default as (None)
> ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'ad_user' in argument_spec defines type as 'str' but documentation doesn't define type
> ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'adfs_authority_url' in argument_spec defines type as 'str' but documentation doesn't define type
> ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'api_profile' in argument_spec defines type as 'str' but documentation doesn't define type
> ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'append_tags' in argument_spec defines type as 'bool' but documentation doesn't define type
> ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'auth_source' in argument_spec defines type as 'str' but documentation doesn't define type
> ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'cert_validation_mode' in argument_spec defines type as 'str' but documentation doesn't define type
> ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'client_id' in argument_spec defines type as 'str' but documentation doesn't define type
> ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'cloud_environment' in argument_spec defines type as 'str' but documentation doesn't define type
> ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'password' in argument_spec defines type as 'str' but documentation doesn't define type
> ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'profile' in argument_spec defines type as 'str' but documentation doesn't define type
> ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'record_type' in argument_spec defines type as 'str' but documentation doesn't define type
> ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'relative_name' in argument_spec defines type as 'str' but documentation doesn't define type
> ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'resource_group' in argument_spec defines type as 'str' but documentation doesn't define type
> ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'secret' in argument_spec defines type as 'str' but documentation doesn't define type
> ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'subscription_id' in argument_spec defines type as 'str' but documentation doesn't define type
> ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'tags' in argument_spec defines type as 'dict' but documentation doesn't define type
> ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'tenant' in argument_spec defines type as 'str' but documentation doesn't define type
> ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'top' in argument_spec defines type as 'int' but documentation doesn't define type
> ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:0:0: parameter-type-not-in-doc: Argument 'zone_name' in argument_spec defines type as 'str' but documentation doesn't define type
> ERROR: plugins/modules/azure_rm_privatednsrecordset_info.py:43:5: documentation-syntax-error: DOCUMENTATION is not valid YAML
> ERROR: tests/sanity/ignore-2.10.txt:725:1: A100: Ignoring 'doc-elements-mismatch' on 'plugins/modules/azure_rm_privatednsrecordset.py' is unnecessar
> ```

Hi @Fred-sun , I have corrected the documentation errors. Could you please check once and provide your feedback? Thanks.

File: tests/sanity/ignore-2.10.txt [hunk 4]
```diff
@@ -727,5 +727,11 @@ plugins/modules/azure_rm_openshiftmanagedcluster.py validate-modules:invalid-ans
 plugins/modules/azure_rm_openshiftmanagedcluster.py validate-modules:required_if-requirements-unknown
 plugins/modules/azure_rm_openshiftmanagedcluster.py validate-modules:parameter-list-no-elements
 plugins/modules/azure_rm_vmbackuppolicy.py validate-modules:parameter-list-no-elements
+plugins/modules/azure_rm_privatednsrecordset.py validate-modules:doc-elements-mismatch
```
Comment (Fred-sun) [inline tests/sanity/ignore-2.10.txt]:
```suggestion

```

File: tests/sanity/ignore-2.11.txt [hunk 5]
```diff
@@ -727,5 +727,11 @@ plugins/modules/azure_rm_openshiftmanagedcluster.py validate-modules:invalid-ans
 plugins/modules/azure_rm_openshiftmanagedcluster.py validate-modules:required_if-requirements-unknown
 plugins/modules/azure_rm_openshiftmanagedcluster.py validate-modules:parameter-list-no-elements
 plugins/modules/azure_rm_vmbackuppolicy.py validate-modules:parameter-list-no-elements
+plugins/modules/azure_rm_privatednsrecordset.py validate-modules:doc-elements-mismatch
```
Comment (Fred-sun) [inline tests/sanity/ignore-2.11.txt]:
```suggestion

```

**Review by Fred-sun — COMMENTED** [review]
fixes sanity errors!

**Comment by aparna-patil** [comment]
@Fred-sun Committed the suggested changes in sanity tests files

**Comment by Fred-sun** [comment]
> @Fred-sun Committed the suggested changes in sanity tests files

All test pass! I will push for merge! Thank you very much!

**Pull request merged**